RecoLocalCalo/EcalRecAlgos/interface/Common.h

```diff
@@ -0,0 +1,48 @@
+#ifndef RecoLocalCalo_EcalRecAlgos_interface_Common_h
+#define RecoLocalCalo_EcalRecAlgos_interface_Common_h
+
+#include <cstdint>
+#include <cmath>
+#include <cassert>
+#include <chrono>
+#include <iostream>
+
+#include <cuda.h>
+#include <cuda_runtime.h>
+
+// a workaround for std::abs not being a constexpr function
+namespace ecal {
+
+template<typename T>
+constexpr T abs(T const& value) {
+    return ::std::max(value, -value);
+}
+
+// temporary
+namespace mgpa {
+
+constexpr int adc(uint16_t sample) { return sample & 0xfff; }
+constexpr int gainId(uint16_t sample) { return (sample>>12) & 0x3; }
+
+}
+
+}
+
+template<typename T>
+struct DurationMeasurer {
```

Comment:
if this is something to be kpt, and not only for debugging, can you add a templated data member to hold the target buffer, instead of `std::cout` ?
This way it could be used with the `MessagerLogger` as well.